rpc/dns_monitor.go

```diff
@@ -0,0 +1,203 @@
+package rpc
+
+import (
+	"context"
+	"net"
+	"sync"
+	"sync/atomic"
+	"time"
+
+	"github.com/sirupsen/logrus"
+)
+
+// DNSMonitor handles background DNS monitoring for worker gateways
+type DNSMonitor struct {
+	enabled       bool
+	checkInterval time.Duration
+	connectionStr string
+	ctx           context.Context
+	cancel        context.CancelFunc
+	stopComplete  chan struct{}
+}
```

Comment:
what about moving all this code into its own pkg? in the end the functionality grew and is not highly attached to rpc, maybe a new pkg inside `/internal` and rpc just instantiates a new dnsMonitor in case is required and pass configs